test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightErrorDeserializerTests.cs

```diff
@@ -114,6 +116,70 @@ public void ReadTopLevelErrorWithCollectionResourceValueInstanceAnnotation_Works
             Assert.Equal("sample value", property.Value);
         }
 
+        [Fact]
+        public void ReadExtendedInnerError()
+        {
+            // Arrange
+            const string payload =
+                    "{\"error\":{" +
+                        "\"code\":\"\",\"message\":\"\"," +
+                        "\"target\":\"any target\"," +
+                        "\"details\":[{\"code\":\"500\",\"target\":\"any target\",\"message\":\"any msg\"}]," +
+                        "\"innererror\":{" +
+                            "\"stacktrace\":\"NormalString\"," +
+                            "\"message\":\"\"," +
+                            "\"type\":\"\"," +
+                            "\"innererror\":{" +

```

Comment:
Maybe add an inner nested object?

Reply:
The MyNewObject is a nested Object. Do you want me to rename it or add another level? 

---
In reply to: [321903391](https://github.com/OData/odata.net/pull/1523#discussion_r321903391) [](ancestors = 321903391)

Reply:
I meant an object nested in MyNewObject.

---
In reply to: [321944035](https://github.com/OData/odata.net/pull/1523#discussion_r321944035) [](ancestors = 321944035,321903391)